src/Reviews.js

```diff
@@ -0,0 +1,79 @@
+import React, { forwardRef, useState } from "react";
+import { FaSearch } from "react-icons/fa";
+import { BsStar, BsStarFill } from "react-icons/bs";
+import reviewsStyling from "./reviewsStyling.module.css";
+import { useContext } from "react";
+import { useParams } from "react-router-dom";
+import { ReviewsContext } from "./App";
+import Review from "./Review";
+import courseContentStyles from "./courseContentStyles.module.css";
+
+const Reviews = forwardRef((props, ref) => {
```

Comment:
https://blog.logrocket.com/build-a-half-star-rating-component-in-react-from-scratch/
check out this article to learn how to create a star review component from scratch. what you did is working fine but I think it could be better.